precompile/testutils/test_predicate.go

```diff
@@ -18,48 +17,27 @@ type PredicateTest struct {
 
 	PredicateContext *precompileconfig.PredicateContext
 
-	StorageSlots [][]byte
-	Gas          uint64
-	GasErr       error
-	PredicateRes []byte
+	PredicateBytes []byte
+	Gas            uint64
+	GasErr         error
+	ExpectedRes    bool
```

Comment:
small suggestion for further improvement - it would be nice to change from returning a boolean to returning an error so that tests can check the exact error value

Reply:
We don't return an error from `VerifyPredicate`. do you mean we should add error to the `VerifyPredicate` output?

Reply:
I'm suggesting that instead of returning a boolean we return an error and treat a non-nil error equivalent to false and vice versa. This would allow us to surface more information about why false was returned and make the tests more explicit.

Reply:
done